pkg/new-ui/v1beta1/README.md

```diff
@@ -0,0 +1,123 @@
+
+We are using [Material UI](https://material.angular.io/) to design frontend. Try to use Material UI components to implement new Katib UI features.
+
+## Folder structure
+
+1. You can find `Dockerfile` and `main.go` - file to serve the UI under `cmd/new-ui/v1beta1`
+
+1. You can find Go backend under `pkg/new-ui/v1beta1`
+
+1. You can find Angular frontend under `pkg/new-ui/v1beta1/frontend`
+
+## Requirements
+
+To make changes to the UI you need to install:
+
+- Tools, defined [here](https://github.com/kubeflow/katib/blob/master/docs/developer-guide.md#requirements).
+
+- `node` (v12 or later) and `npm` (v6.13 or later). Recommended to install `node` and `npm` using [`nvm`](https://github.com/nvm-sh/nvm). After installing `nvm`, you can run `nvm install 12.18.1` to install `node` version 12.18.1 and run `nvm use 12.18.1` to use that version.
+
+## Development
+
+While development you have different ways to run Katib UI.
+1. Build and serve only the frontend. The dev server will also be proxying requests to the backend
+2. Build the frontend and serve it via the backend locally
+
+### Serve only the frontend
+
+You can run a webpack dev server that only exposes the frontend files, which can be useful for testing only the UI of the app. There's also a `proxy.conf.json` file which configures the dev server to send the backend requests to port `8000`.
+
+In order to build the UI locally, and expose it with a webpack dev server you will need to:
+1. Create a module from the [common library](https://github.com/kubeflow/kubeflow/tree/master/components/crud-web-apps/common/frontend/kubeflow-common-lib)
+2. Install the node modules of the app and also link the common-library module
+
+
+
+You can build the common library with:
+```bash
+cd /tmp && git clone https://github.com/kubeflow/kubeflow.git \
+  && cd kubeflow \
+  && git checkout a349284 \
+  && cd components/crud-web-apps/common/frontend/kubeflow-common-lib
+
+# build the common library module
+npm i
+npm run build
+
+# link the module to your npm packages
+# depending on where you npm stores the global packages you
+# might need to use sudo
+npm link dist/kubeflow
+```
+
+And then build and run the UI locally, on `localhost:4200`, with:
+```bash
+# If you've already cloned the repo then skip this step and just
+# navigate to the pkg/new-ui/v1beta1/frontend dir
+cd /tmp && git clone https://github.com/kubeflow/katib.git \
+  && cd katib/pkg/new-ui/v1beta1/frontend
+
+npm i
+npm link kubeflow
+npm run start
+```
+
+### Serve the UI  from the backend
+
+This is the recommended way to test the web app e2e. In order to build the UI and serve it via the backend, locally, you will need to:
+
+1. Build the UI locally. You can follow the steps from the previous section, but instead of running `npm run start` you will need to run `npm run build:watch`. This will start a process that will be watching the source code and building the frontend artifacts under `frontend/dist/static` folder.
```

Comment:
What do you think about this ?
```suggestion
1. Build the UI locally. You can follow the steps from the previous section, but instead of running `npm run start` you have to run `npm run build`. That creates build under `funder/dist/default` folder. 
 
Moreover, instead of `npm run build` you can run `npm run build:watch` which starts a process that is watching the source code and building the frontend artefacts under `frontend/dist/default` folder.

Learn more about Angular scripts in the [official guide](https://angular.io/cli/build).
```

Reply:
I think there's a typo there in the word `funder` [should be `frontent`] right?

Also we must use the `static` folder for the produced files, as we just discussed https://github.com/kubeflow/katib/pull/1427#discussion_r580918723.

Reply:
> I think there's a typo there in the word funder [should be frontent] right?

You are right, sorry 👍 

> Also we must use the static folder for the produced files, as we just discussed #1427 (comment).

Got it.

Reply:
[ lol, I made a typo myself when I asked if you had made a typo `frontent` -> `frontend` ]

I'll create a new commit on top with your proposed changes there

Reply:
I can't put the dedicated paragraph for `build:watch` in the proposed place since it will break the numbered list.

@andreyvelich could you make another proposed commit with how you would like us to phrase this?

Reply:
Sure, let me check.